### PR TITLE
Issue #166 add more fields to signup form

### PR DESCRIPTION
**Pull request by abulojoshua1**
### What Does The PR Do?
- Adds extra fields into the signup form

### Changes Made:
- Added a `request_to_join_group` many-to-many filed into IHPProfile that maps to `groups.GroupProfile` model
- I exposed `company`, `position`, `country`, and `request_to_join_group` to the signup form with `request_to_join_group` as an optional multi-select form using select2
- Saved the form fields on form submission

### Screenshot
Screenshot 2020-03-27 at 11 22 34

### CODE REVIEW

**Comment by giohappy** [comment]
For the selection of groups the select2 widget already used in other parts of GeoNode should be used.
Take for example the Metadta Wizard (you need to have a Document or a Layer loaded to see it).

You shoudl use the GroupProfile API to retrieve the groups. **Only public grouos must be listed!**

image


image

File: ihp/content/forms.py [hunk 1]
```diff
@@ -41,6 +43,19 @@ def _replace_username_with_first_last(form):
...
         widget=forms.TextInput(attrs={'placeholder': _("Name of the person")}))
+
+    fields["organization"] = forms.CharField(min_length=2, label=_(u"Organization"),
+        widget=forms.TextInput(attrs={'placeholder': _(u"Organization")}), required=False)
+    fields["position"] = forms.CharField(min_length=2, label=_(u"Position"),
+        widget=forms.TextInput(attrs={'placeholder': _(u"Position")}), required=False)
+    fields["country"] = forms.CharField(min_length=2, label=_(u"Country"),
+        widget=forms.TextInput(attrs={'placeholder': _(u"Country")}), required=False)
+
+    fields["request_to_join_group"] = TagField(
```
Comment (petrus7) [inline ihp/content/forms.py]: Isnt it cleaner to declare is as form field explicitly ?

File: ihp/people/tests/test_signup.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+from django.test import Client, TestCase
+from django.urls import reverse
+
+from ihp.people.models import IHPProfile
+from geonode.groups.models import GroupProfile
+from django.contrib.auth.models import Group
+
+
+@pytest.mark.django_db
+class TestUserRegistration(TestCase):
```
Comment (petrus7) [inline ihp/people/tests/test_signup.py]: Id focus on testing our custom functionalities not allauth

File: ihp/content/forms.py (hunk 1 above)
Comment (abulojoshua1) [reply ihp/content/forms.py]: @petrus7, I tried a couple of form fields but all did not seem to work well with the select2 js library. On form submission, the values do not seem to arrive as a list of user-selected values throwing an error (i.e, "Enter a list of values.").

File: ihp/people/tests/test_signup.py (hunk 2 above)
Comment (abulojoshua1) [reply ihp/people/tests/test_signup.py]: :) Those test code in the forms and adapters, not allauth.

Comment (giohappy) [reply ihp/people/tests/test_signup.py]: @abulojoshua1 you made a lot of registration tests, but I don't see a test for login with email

Comment (abulojoshua1) [reply ihp/people/tests/test_signup.py]: @giohappy, the login tests are in [this PR](https://github.com/geosolutions-it/unesco-ihp/pull/214).

**Pull request merged**